### PR TITLE
Make useSocket singleton-ish + improve analytics

### PR DESCRIPTION
**Pull request by vudung45**
- Implement a backend event handler to emit the list of rooms the user is currently in and use that for chatAnalytics
- make useSocket() hook singleton. useSocket() can now be called globally without creating new socket.
- migrate from `react-cookie` to `universal-cookie` since it's deprecated.

Update to adapt changes from @namdaoduy's latest PR:
User view (dropdown item is not clickable):
Screen Shot 2020-09-01 at 12 16 23 AM

Admin view (dropdown is clickable and will switch room onClick):
Screen Shot 2020-09-01 at 12 05 55 AM

### CODE REVIEW

File: frontend/src/Hooks/Analytics.tsx [hunk 1]
```diff
@@ -30,22 +35,35 @@ function useChatRoomsAnalyticsImpl() {
           numUsers: r.response.map((r) => r.count).reduce((a, b) => a + b),
           numRooms: r.response.length,
           rooms: r.response,
+          currentRoomNames: analytics.currentRoomNames
         };
         setAnalytics(newAnalytics);
-        if (runRoutine) setTimeout(() => updateRoutine(), 5000);
+        if (runRoutine) setTimeout(() => updateRoutine(), 2000);
```
Comment (tuanh118) [inline frontend/src/Hooks/Analytics.tsx:41]: How expensive is this request for server?

Comment (vudung45) [reply frontend/src/Hooks/Analytics.tsx:41]: Unknown :) , but we can set back to 5000 if needed

Comment (tuanh118) [reply frontend/src/Hooks/Analytics.tsx:41]: yeah we should be good at 5000 or even higher to avoid getting hammered by this request

Comment (manhhung741) [reply frontend/src/Hooks/Analytics.tsx:41]: anh co suggest la send cai room info khi ma co' ng moi join hay leave thoi

Comment (vudung45) [reply frontend/src/Hooks/Analytics.tsx:41]: @manhhung741  will address this in later commit

**Pull request merged**